src/index/hnsw/faiss_hnsw_config.h

```diff
@@ -43,9 +43,10 @@ class FaissHnswConfig : public BaseHnswConfig {
             .for_static();
         KNOWHERE_CONFIG_DECLARE_FIELD(refine_k)
             .description("refine k")
-            .allow_empty_without_default()
+            .set_default(1)
             .set_range(1, std::numeric_limits<CFG_FLOAT::value_type>::max())
-            .for_search();
+            .for_search()
```

Comment:
for_iterator?

Reply:
removed `.for_range_search()`. Technically, `range_k` is applicable only for the search, neither range_search, not iterator.